target/scripts/startup/setup.d/getmail.sh

```diff
@@ -4,38 +4,46 @@ function _setup_getmail() {
   if [[ ${ENABLE_GETMAIL} -eq 1 ]]; then
     _log 'trace' 'Preparing Getmail configuration'
 
-    local GETMAILRC ID CONFIGS
+    local GETMAIL_RC ID GETMAIL_DIR
 
-    GETMAILRC='/etc/getmailrc.d'
-    CONFIGS=0
+    local GETMAIL_CONFIG_DIR='/tmp/docker-mailserver/getmail'
+    local GETMAIL_RC_DIR='/etc/getmailrc.d'
+    local GETMAIL_RC_GENERAL_CF="${GETMAIL_CONFIG_DIR}/getmailrc_general.cf"
+    local GETMAIL_RC_GENERAL='/etc/getmailrc_general'
 
-    mkdir -p "${GETMAILRC}"
+    # Create the directory /etc/getmailrc.d to place the user config in later.
+    mkdir -p "${GETMAIL_RC_DIR}"
 
-    # Generate getmailrc configs, starting with the `/etc/getmailrc_general` base config,
-    # Add a unique `message_log` config, then append users own config to the end.
-    for FILE in /tmp/docker-mailserver/getmail-*.cf; do
-      if [[ -f ${FILE} ]]; then
-        CONFIGS=1
-        ID=$(cut -d '-' -f 3 <<< "${FILE}" | cut -d '.' -f 1)
-        local GETMAIL_CONFIG="${GETMAILRC}/getmailrc-${ID}"
+    # Check if custom getmailrc_general.cf file is present.
+    if [[ -f "${GETMAIL_RC_GENERAL_CF}" ]]; then
+      _log 'debug' "Custom 'getmailrc_general.cf' found"
+      cp "${GETMAIL_RC_GENERAL_CF}" "${GETMAIL_RC_GENERAL}"
+    fi
 
-        cat /etc/getmailrc_general                                 >"${GETMAIL_CONFIG}"
-        echo -e "message_log = /var/log/mail/getmail-${ID}.log\n" >>"${GETMAIL_CONFIG}"
-        cat "${FILE}"                                             >>"${GETMAIL_CONFIG}"
-      fi
-    done
+    # If no matching filenames are found, and the shell option nullglob is disabled, the word is left unchanged.
+    # If the nullglob option is set, and no matches are found, the word is removed.
+    shopt -s nullglob
 
-    if [[ ${CONFIGS} -eq 1 ]]; then
-      cat >/etc/cron.d/getmail << EOF
-*/${GETMAIL_POLL} * * * * root /usr/local/bin/getmail-cron
-EOF
-      chmod -R 600 "${GETMAILRC}"
-    fi
+    # Generate getmailrc configs, starting with the `/etc/getmailrc_general` base config, then appending users own config to the end.
+    for FILE in "${GETMAIL_CONFIG_DIR}"/*.cf; do
+      if [[ ${FILE} =~ /getmail/(.+)\.cf && ${FILE} != "${GETMAIL_RC_GENERAL_CF}" ]]; then
```

Comment:
Again, we usually use `[[ ... ]] && [[ ... ]]`.

Reply:
BTW: What is the rational in this project to quote variables only on the right side of a condition check?

`[[ ${foo} == "${bar}" ]]` is the same as `[[ ${foo} == ${bar} ]]`  or `[[ "${foo}" == ${bar} ]] `

Reply:
> BTW: Do you know, what the rational in this project is to quote variables only on the right side of a condition check?
> 
> `[[ ${foo} == "${bar}" ]]` is the same as `[[ ${foo} == ${bar} ]]`  or `[[ "${foo}" == ${bar} ]] `

It's actually not: <https://github.com/koalaman/shellcheck/wiki/SC2053>

Reply:
Thanks for clarification 👍

So best practice is still, quoting every variable. Then you are safe. In the example above, it feels odd to just quote the right side.

Reply:
Funnily enough, there are places where quoting does not seem to be necessary, which I think is extremely odd. For example: variable assignments to other variables' values or the LHS of such `if` statements. I don't know why there is a difference between the LHS and RHS.

You are correct, though: quoting is always the safest way of doing it.

Reply:
The review discussions for this PR has been a great example of why I'm averse to heavy use of shell scripts to manage a project 😆 

Plenty of this stuff we shouldn't even need to be thinking about or questioning with something like rust 😩